File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Model/Entities/MassimariScarto/MassimarioScartoModel.cs
using System;

namespace VecompSoftware.DocSuiteWeb.Model.Entities.MassimariScarto
{
    public class MassimarioScartoModel
    {
        #region [Constructor]

        public MassimarioScartoModel() { }

        public MassimarioScartoModel(Guid uniqueId)
        {
            UniqueId = uniqueId;
        }

        #endregion

        #region [Properties]

        public Guid? UniqueId { get; set; }

        public short? Code { get; set; }

        public string FullCode { get; set; }

        public short? ConservationPeriod { get; set; }

        public DateTimeOffset? EndDate { get; set; }

        public DateTimeOffset StartDate { get; set; }

        public MassimarioScartoStatus Status { get; set; }

        public short MassimarioScartoLevel { get; set; }

        public string MassimarioScartoPath { get; set; }

        public string MassimarioScartoParentPath { get; set; }

        public string Name { get; set; }

        public string Note { get; set; }

        public Guid? FakeInsertId { get; set; }

        public DateTimeOffset? RegistrationDate { get; set; }

        public string RegistrationUser { get; set; }

        public DateTimeOffset? LastChangedDate { get; set; }

        public string LastChangedUser { get; set; }

        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Gui/App/Models/MassimariScarto/MassimarioScartoModel.ts
import MassimarioScartoStatusType = require('App/Models/MassimariScarto/MassimarioScartoStatusType');

class MassimarioScartoModel {
    UniqueId: string;
    Status: MassimarioScartoStatusType;
    Name: string;
    Code: number;
    FullCode: string;
    Note: string;
    ConservationPeriod: number;
    StartDate: Date;
    EndDate: Date;
    MassimarioScartoPath: string;
    MassimarioScartoParentPath: string;
    MassimarioScartoLevel: number;
    FakeInsertId: string;
    RegistrationUser: string;
    RegistrationDate: Date;
    LastChangedUser: string;
    LastChangedDate: Date;

    constructor() {       
    }

    isActive(): boolean {
        return this.EndDate == undefined || moment(this.EndDate).isAfter(moment().toDate());
    }

    getPeriodLabel(): string {
        let label: string = '';
        switch (this.ConservationPeriod) {
            case -1:
                label = 'Illimitato';
                break;
            default:
                label = this.ConservationPeriod.toString().concat(' Ann', this.ConservationPeriod == 1 ? 'o' : 'i');
                break;
        }
        return label;
    }
}

export = MassimarioScartoModel;